### ISSUE TITLE
xbox360ce hangs (doesn't respond) when an xbox one controller is connected

### ISSUE THREAD

**Issue by DanielTrifonov** [x360ce/x360ce#378]
As I said in the title above the application stops responding as soon as it registers that an xbox one controller is connected. Tests show that as long as the controller is not connected, the application is working alright.

Process dump in the link below:

https://www.dropbox.com/s/ufy3rfqontijo9r/x360ce_x64%20%282%29.DMP?dl=0

**Comment by DanielTrifonov** [comment]
After further testing the application seems to hang up after connecting two controllers. Any ideas on how to fix it would be welcome

**Comment by abuteler** [comment]
Having the same issue.
I have a PS2 controller with a PC adapter and an XBox One controller, and the GUI hangs after driver installation finishes. Any click leaves the program unresponsive. :(

**Comment by abuteler** [comment]
(if I do the installation with my xbox one controller unplugged, the program does what's expected just fine.)

**Comment by TawRex** [comment]
Hi there,
I am also experiencing this problem.  I had not run GTA-V for a week or more. Now, when starting X360CE_X64 would cause GTA-V to hang during startup.

Further investigation showed that with two controllers connected (A G27 and an XBOX One controller for PC) the X360CE GUI would hang after detecting the hardware. Additionally, for some reason, X360CE treated my XBOX-One controller as new, with a new GUID, I'm not sure just why that occurred.

Anyway, bottom line is something related to the XBOX-One controller has broken X360CE.  The program will go unresponsive and have to be ended forcefully.  Unless I disconnect the XBOX-One controller, at which point the program becomes responsive again.

No doubt some recent Microsoft driver change?  This combination used to work perfectly on my system.

Cheers.

**Comment by TawRex** [comment]
I don't know if it may be related, but both xboxgip.sys and xinputhid.sys have been updated by Microsoft recently.  Unfortunately I did not have a backup of my system prior to the 5th of March, and these newer drivers were in place then.

Perhaps some change that has broken x360ce.  I certainly hope you are able to have a look at this issue, it's a real showstopper.

Thanks

**Comment by Senryoku** [comment]
Hello,

Just reporting the same problem trying to use two different controllers (XBox One pad and a Mayflash arcade stick). With either one alone everything seems to work just fine, but if both are connected when launching x360ce_x64.exe, it hangs.
As TawRex mentioned, this is probably due to some recent change outside of x360ce as this exact same setup was working fine for me a month ago.

Thanks!

**Comment by TawRex** [comment]
So, I decided to download the source package, build it and have a go at running X360CE under debug. Both the x86 and x64 builds display this same issue. 

The program goes into a tight loop in some 'External' code, from what I can determine. If I force VS2013 to break into the program, whilst in the loop, it always stops in the following code block in 'XInput.cs' at the line "return method(dwUserIndex, out pState);".  Normally, with only one controller connected, the program passes through this code repetitively waiting for input operations.  For some reason, when the XBOX One controller is connected with my G27 Wheel, the call to the method resolved by "method = GetMethod<XInputGetStateDelegate>(functionName)", where functionName is 'XInputGetState" will fail, or hang ???

```
    public static unsafe ErrorCode XInputGetState(int dwUserIndex, out State pState)
    {
        var functionName = "XInputGetState";
        //if (IsGetStateExSupported) functionName = "XInputGetStateEx";
        pState = new State();
        try
        {
            var method = GetMethod<XInputGetStateDelegate>(functionName);
            return method(dwUserIndex, out pState);
        }
        catch (AccessViolationException ex) { throw new Exception(ex.Message); }
        catch (Exception) { throw; }
    }
```

Unfortunately my skills with C# and this type of code is not up to resolving just what is wrong. Although I shall keep looking.

We need help.

**Comment by Squall-Leonhart** [comment]
This is caused by a windows 10 update, and is not a fault in our code at all.

**Comment by TawRex** [comment]
Thanks for the response Sqall-Leonhart.  I understand that this is due to a Microsoft change in the system driver XINPUTHID.SYS  However, that does not change the fact that this has broken the function of X360CE.

Are you saying that because the fault is not in X360CE that you will not investigate, or fix it ?  Or is it a case of it cannot be fixed because there is a bug in the new version of the XInput driver? 

Thanks

**Comment by Squall-Leonhart** [comment]
The bug occurs even without X360CE, there are reports of it across several microsoft answers threaes.

**Comment by TawRex** [comment]
Thanks for that mate.  I must admit I had not looked for other related issues on MS sites.

OK, for those of you, like me, that find this bug in the new XInput driver from Microsoft a show stopper. Here is what I did to temporarily work around the issue.

First, look in "C:\Windows\System32\drivers" for the file "XINPUTHID.SYS"  Right click on the file and check its properties.  If the version number is 10.0.10586.122 then you have the bad version.

What I have done is to extract this driver from the install.wim file on my Windows 10 upgrade DVD that I burnt from the ISO downloaded from Microsoft.  The version on the 10586 upgrade is 10.0.10586.0  This is the version we need to work around this problem.

I then took ownership of the "XINPUTHID.SYS" file (actually it is a symbolic link) and renamed it.  I then placed the old version of the same file that I had extracted from the install.wim into "C:\Windows\System32\drivers"

After restarting the system, I plugged in my XBOX One controller and checked device manager. Sure enough it had loaded the older version of the driver.

I then ran X360CE_X64 in my GTA5 directory, with both my G27 and the XBOX controller connected.  Everything worked correctly.  X360CE worked as expected, and GTA5 loaded and ran with input from both controller being combined.  Yay!!!

Not an ideal solution. I will have to keep a close eye on the file XINPUTHID.SYS incase Windows overwrites it.  Hopefully with a newer, and better version in the not too distant future.

Good luck to you all.

**Comment by TawRex** [comment]
Last time I'll trouble you all with this.

After Squall-Leonhart's tip that this was a known Microsoft problem I searched more widely, and sure enough this seems to be quite an issue. The dodgy drivers were introduced by a Cumulative update - KB3140743.

The best way to fix this issue, until Microsoft do something about it, is to revert the drivers to a previous version. An earlier driver pack is available from the Windows Update Catalog (for Windows 7/8/8.1).

Check out the descriptions provided here: 
http://forum.halomaps.org/index.cfm?page=topic&topicID=49128

Cheers.

**Comment by DanielTrifonov** [comment]
I found a workaround. It includes modifying x360ce.ini. First, connect your first controller and then run the x360ce.exe. When it imports the settings from the internet set you controls, save and close the program. Then disconnect the first controller and connect the second one. Set it up and save and close the program. Then open the x360ce.ini and look for the two instances of these IDs ([IG_009ee420e1ed11e58001444553540000] <--- something like this). Copy the first one (without the [ ] brackets) and paste it in the [Mappings] section in PAD1=. Repeat it again for the second controller by looking for the other ID and pasting it (again, without the brackets) in PAD2=. Connect both controllers, paste the .ini file along with the other files in the chosen game dir and enjoy playing.

**Comment by Nucleoprotein** [comment]
I do not think MS will fix this issue, because new drivers include some changes for their Universal Apps in shop. Update also blocked XBOX Guide Buttons in SDL2 so apps like Steam cannot use it. Old fix ie. removing Xbox App does not work, you need to login in Xbox App and disable DVR.

**Comment by Squall-Leonhart** [comment]
they will have to, since it is breaking keyboards, mice, wheels, etc.

**Comment by datdudequick** [comment]
@DanielTrifonov I'm also having the same problem. Tried your fix and was still unsuccessful at getting both controllers to work for local SF5 play.

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
